src/components/community/community-base.js

```diff
@@ -0,0 +1,22 @@
+import { communityUrlFactory } from './community-url-factory.js';
+import { LanguageListenerController } from '../../controllers/language-listener/language-listener.js';
+import { LitElement } from 'lit';
+
+export class CommunityBase extends LitElement {
+
+	static get properties() {
+		return {
+			articleMap: { attribute: 'article-map', type: Object }
+		};
+	}
+
+	constructor() {
+		super();
+		this.langController = new LanguageListenerController(this);
+	}
+
+	connectedCallback() {
+		super.connectedCallback();
+		this.communityArticleDirective = communityUrlFactory(this.articleMap);
```

Comment:
I think this is going to result in the map getting resolved on first connection and if the consumer changes `articleMap` (via property or attribute) it will never get updated. To fix, you could either do this in `updated` whenever `articleMap` changes?